agent/backend/cmd.go

```diff
@@ -0,0 +1,90 @@
+package backend
+
+import (
+	"github.com/go-cmd/cmd"
+)
+
+// CmdInterface abstracts the functionality from go-cmd/cmd package
+type CmdInterface interface {
```

Comment:
This name doesn't seem to follow Go idiomatic naming conventions https://go.dev/doc/effective_go#interface-names, how about `Commander`?